src/Microsoft.Health.Fhir.Shared.Api/Features/Resources/Bundle/TransactionBundleValidator.cs

```diff
@@ -68,50 +70,47 @@ private static string BuildRequestUrlForConditionalQueries(EntryComponent entry,
             if (string.IsNullOrWhiteSpace(entry.Request.IfNoneExist) && !entry.Request.Url.Contains("?", StringComparison.Ordinal))
             {
-                if (entry.Request.Method == HTTPVerb.POST)
-                {
-                    return entry.FullUrl;
-                }
+                return entry.Request.Method == HTTPVerb.POST ? entry.FullUrl : entry.Request.Url;
+            }
+
+            string resourceType = null;
+            StringValues conditionalQueries;
 
-                return entry.Request.Url;
+            if (entry.Request.Method == HTTPVerb.PUT)
+            {
+                string[] conditionalUpdate = entry.Request.Url.Split('?');
+                resourceType = conditionalUpdate[0];
+                conditionalQueries = conditionalUpdate[1];
             }
-            else
+
+            if (entry.Request.Method == HTTPVerb.POST)
             {
-                string resourceType = null;
-                StringValues conditionalQueries;
-                HTTPVerb requestMethod = (HTTPVerb)entry.Request.Method;
-                bool conditionalCreate = requestMethod == HTTPVerb.POST;
-                bool condtionalUpdate = requestMethod == HTTPVerb.PUT;
+                resourceType = entry.Request.Url;
+                conditionalQueries = entry.Request.IfNoneExist;
+            }
 
-                if (condtionalUpdate)
-                {
-                    string[] conditinalUpdateParameters = entry.Request.Url.Split("?");
-                    resourceType = conditinalUpdateParameters[0];
-                    conditionalQueries = conditinalUpdateParameters[1];
-                }
-                else if (conditionalCreate)
-                {
-                    resourceType = entry.Request.Url;
-                    conditionalQueries = entry.Request.IfNoneExist;
-                }
+            IReadOnlyCollection<SearchResultEntry> matchedResults = await _referenceResolver.GetExistingResourceId(entry.Request.Url, resourceType, conditionalQueries, cancellationToken);
 
-                IReadOnlyCollection<SearchResultEntry> matchedResults = await _referenceResolver.GetExistingResourceId(entry.Request.Url, resourceType, conditionalQueries, cancellationToken);
-                int? count = matchedResults?.Count;
+            if (matchedResults?.Count > 1)
+            {
+                // Multiple matches: The server returns a 412 Precondition Failed error indicating the client's criteria were not selective enough
+                throw new PreconditionFailedException(string.Format(Api.Resources.ConditionalOperationInBundleNotSelectiveEnough, conditionalQueries));
+            }
 
-                if (count > 1)
+            if (matchedResults?.Count == 1)
+            {
+                // If this entry has a fullUrl, then save it to the idDictionary for matching later
+                if (!string.IsNullOrWhiteSpace(entry.FullUrl))
```

Comment:
does something validate that the FullUrl starts with `urn:oid:` in this case somewhere?

Reply:
No. There's no requirement that the `FullUrl` must start with that, it must only be a valid `URI` - https://www.hl7.org/fhir/datatypes.html#uri.